code/game/objects/items/stacks/cable_coil.dm

```diff
@@ -304,34 +305,28 @@
 	. = ..()
 	color = pick(COLOR_RED, COLOR_BLUE, COLOR_GREEN, COLOR_WHITE, COLOR_MAGENTA, COLOR_YELLOW, COLOR_CYAN)
 
-/obj/item/stack/cable_coil/attack(mob/M as mob, mob/user as mob)
-	if(ishuman(M))
-		var/mob/living/carbon/human/H = M
-
-		var/obj/limb/S = H.get_limb(user.zone_selected)
-		if(!(S.status & (LIMB_ROBOT|LIMB_SYNTHSKIN)) || user.a_intent != INTENT_HELP)
+/obj/item/stack/cable_coil/attack(mob/target_mob as mob, mob/user as mob)
+	if(ishuman(target_mob))
+		var/mob/living/carbon/human/humanus = target_mob
+		var/obj/limb/target_limb = humanus.get_limb(user.zone_selected)
+		if(!(target_limb.status & (LIMB_ROBOT|LIMB_SYNTHSKIN)) || user.a_intent != INTENT_HELP)
 			return ..()
-
 		if(user.action_busy)
 			return
-		var/self_fixing = FALSE
-
-		if(H.species.flags & IS_SYNTHETIC && M == user)
-			self_fixing = TRUE
-
-		if(S.burn_dam > 0 && use(1))
-			if(self_fixing)
-				user.visible_message(SPAN_WARNING("\The [user] begins fixing some burn damage on their [S.display_name]."), \
-					SPAN_WARNING("You begin to carefully patch some burn damage on your [S.display_name] so as not to void your warranty."))
-				if(!do_after(user, 30, INTERRUPT_ALL, BUSY_ICON_FRIENDLY))
-					return
-
-			S.heal_damage(0, 15, TRUE)
-			H.pain.recalculate_pain()
-			user.visible_message(SPAN_DANGER("\The [user] repairs some burn damage on \the [M]'s [S.display_name] with \the [src]."))
+		if(target_limb.burn_dam && use(5))
+			if(user == humanus)
+				user.visible_message(SPAN_WARNING("\The [user] begins fixing some burns on from its [target_limb.display_name]."), \
+				SPAN_WARNING("You try to patch some burns on your [target_limb.display_name] by yourself"))
+				healing_time += 3 SECONDS
+			if(!do_after(user, max(healing_time - (user.skills.get_skill_level(SKILL_ENGINEER)SECONDS)), INTERRUPT_ALL, BUSY_ICON_FRIENDLY))
+				return
```

Comment:
As a physical item being used, the cable used should be dropped on the ground on do_after interruption.

Alternatively, only check the amount before hand, and only consume it (and confirm it could be consumed) after the do_after.